gapic-generator-cloud/templates/cloud/client_test.erb

```diff
@@ -170,48 +78,85 @@ describe Google::Cloud::Speech::V1::SpeechClient do
       }
       uri = "gs://bucket_name/file_name.flac"
       audio = { uri: uri }
+<% end %>
 
       # Create expected grpc response
       expected_response = {}
       expected_response = Google::Gax.to_proto expected_response,
-                                               Google::Cloud::Speech::V1::LongRunningRecognizeResponse
+                                               <%= ruby_namespace namespaces %>::<%= method_name.camelize %>Response
```

Comment:
I'm running into difficulty with this one, because the method proto does not describe the nested response (`Google::Cloud::Speech::V1::LongRunningRecognizeResponse`), only the `output_type=".google.longrunning.Operation" `.

Well, this isn't entirely true: The proto contains:

```
leading_comments=" Performs asynchronous speech recognition: receive results via the\n google.longrunning.Operations interface. Returns either an\n `Operation.error` or an `Operation.response` which contains\n a `LongRunningRecognizeResponse` message.\n"
```

Reply:
This is covered by the new config spec (you can check the doc linked from our meeting invite or ping me and I'll send you a link).

However, to cause more confusion, the implementation of those configuration options has changed a bit in the past couple of weeks and the docs have not been kept up to date.

On the plus side, it's not too complicated. Here's a proto that I use for tests in another generator that I think will help you understand all the available options (this covers almost everything with the exception of the oauth scopes and resource names):
https://github.com/googleapis/gapic-generator-kotlin/blob/master/generator/src/test/resources/proto/google/example/test_annotations.proto

I think it'll really help if we can get the test infrastructure @blowmage is working on out. Then we can double check that we have all these configuration options covered in our "garbage" protos and that we all understand them (also there are some gaps - fortunately this case isn't one of them).

So, to generate this particular method, we would expect the proto to have this annotation on it:
https://github.com/googleapis/gapic-generator-kotlin/blob/72b36a26fd1dc2f03aa4c82697203bd86d66bf66/generator/src/test/resources/proto/google/example/test_annotations.proto#L51

If it has that annotation you can extract the type info that you need to generate this method and document the return type correctly. If it does not have that annotation then the method should be treated as a regular method and simply return a `Google::Longrunning.Operation` with no special treatment.

Finally, please keep in mind that that this whole configuration thing is still in flux. If you check our `.gitmodules` we have a dependency on the protos that defines the configuration at `gapic-generator/api-common-protos`, which is now out of date.